ssz/sedes/bitlist.py

```diff
@@ -59,6 +59,13 @@ def deserialize(self, data: bytes) -> Tuple[bool, ...]:
         as_integer = int.from_bytes(data, "little")
         len_value = get_bitlist_len(as_integer)
 
+        # Last byte in bytes should be >= 1, if not data is not a serialised bitlist.
+        if data[len(data) - 1] == 0:
```

Comment:
A slightly more canonical way to do this is via `if data[-1] == 0`

However, we might have an issue with this check raising an `IndexError` in the event of being passed `data = b''`.  We can prevent this by changing the check to be if `data[-1:] == b'\x00'`.  It would be good to add a new test to cover this case since it appears we don't have anything checking that currently.

Reply:
I added a test to check that the len(data) >= 1 otherwise it cannot be a serialisation of a bitlist.
As the test for the last. byte comes later, we can prove that the data[-1] will never raise an IndexError. Indeed, the code looks similar to our Dafny code for [this function](https://github.com/PegaSysEng/eth2.0-dafny/blob/6eb5f93a2ce7a4d6f436ae5c08a720feddf99c53/src/dafny/ssz/BitListSeDes.dfy#L119) and the two preconditions `len(data) >= 1` and `data[-1] != 0` are enough to make sure the `data` has the form of a serialised bitlist (proved in this [lemma](https://github.com/PegaSysEng/eth2.0-dafny/blob/6eb5f93a2ce7a4d6f436ae5c08a720feddf99c53/src/dafny/ssz/BitListSeDes.dfy#L289). 

I also added a test to check that the exception is raised on an empty `data`.

Reply:
@pipermerriam Re formatting with black, I have installed and reformatted the files, but still get an error when I run `make lint-roll`:
```
franck:~/development/franck44-py-ssz [master]$ black tests/sedes/test_bitlist_serializer.py
All done! ✨ 🍰 ✨
1 file left unchanged.
franck:~/development/franck44-py-ssz [master]$ make lint-roll
isort --recursive ssz tests
/Library/Developer/CommandLineTools/usr/bin/make lint
tox -elint
lint develop-inst-noop: /Users/franck/development/franck44-py-ssz
lint installed: appdirs==1.4.4,attrs==19.3.0,black==19.3b0,click==7.1.2,cytoolz==0.10.1,entrypoints==0.3,eth-hash==0.2.0,eth-typing==2.2.1,eth-utils==1.9.0,flake8==3.7.8,isort==4.3.21,lru-dict==1.1.6,mccabe==0.6.1,pycodestyle==2.5.0,pyflakes==2.1.1,pyrsistent==0.16.0,six==1.14.0,-e git+git@github.com:franck44/py-ssz.git@cda99a64d68ced1ee8c051f4acd87244de5f62b5#egg=ssz,toml==0.10.1,toolz==0.10.0
lint run-test-pre: PYTHONHASHSEED='2896134891'
lint run-test: commands[0] | flake8 /Users/franck/development/franck44-py-ssz/ssz /Users/franck/development/franck44-py-ssz/tests /Users/franck/development/franck44-py-ssz/scripts
lint run-test: commands[1] | black --check /Users/franck/development/franck44-py-ssz/ssz /Users/franck/development/franck44-py-ssz/tests /Users/franck/development/franck44-py-ssz/scripts
would reformat /Users/franck/development/franck44-py-ssz/tests/sedes/test_bitlist_serializer.py
All done! 💥 💔 💥
1 file would be reformatted, 64 files would be left unchanged.
ERROR: InvocationError for command /Users/franck/development/franck44-py-ssz/.tox/lint/bin/black --check ssz tests scripts (exited with code 1)
```
Any idea what can go wrong? I am not a Python/PyCharm regular user :-(